src/test/java/htsjdk/samtools/cram/build/Sam2CramRecordFactoryTest.java

```diff
@@ -0,0 +1,171 @@
+package htsjdk.samtools.cram.build;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.cram.common.CramVersions;
+import htsjdk.samtools.cram.encoding.readfeatures.Bases;
+import htsjdk.samtools.cram.encoding.readfeatures.ReadBase;
+import htsjdk.samtools.cram.encoding.readfeatures.ReadFeature;
+import htsjdk.samtools.cram.encoding.readfeatures.Substitution;
+import htsjdk.samtools.cram.structure.CramCompressionRecord;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by vadim on 06/06/2017.
+ */
+public class Sam2CramRecordFactoryTest {
+
+    @Test
+    public void testIsACGTN() {
+        for (byte base = Byte.MIN_VALUE; base < Byte.MAX_VALUE; base++) {
+            if (base == 'A' || base == 'C' || base == 'G' || base == 'T' || base == 'N') {
+                Assert.assertTrue(Sam2CramRecordFactory.isACGTN(base));
+            } else {
+                Assert.assertFalse(Sam2CramRecordFactory.isACGTN(base));
+            }
+        }
+    }
+
+    /**
+     * Test the outcome of a matching base.
+     * The result should be empty - no base read features issued.
+     */
+    @Test
+    public void testAddSubstitutionsAndMaskedBasesMatch() {
+        final byte[] refBases = "A".getBytes();
+        final byte[] readBases = "A".getBytes();
+        final byte[] scores = "!".getBytes();
+
+        final SAMFileHeader header = new SAMFileHeader();
+        final CramCompressionRecord record = new CramCompressionRecord();
+        record.alignmentStart = 1;
+        final List<ReadFeature> readFeatures = new ArrayList<>();
+        final int fromPosInRead = 0;
+        final int alignmentStartOffset = 0;
+        final int nofReadBases = 1;
+
+        final Sam2CramRecordFactory s2mFactory_v2 = new Sam2CramRecordFactory(refBases, header, CramVersions.CRAM_v2_1);
```

Comment:
I'm not sure we should maintain tests or code for creating CRAM 2.1 - I don't think its possible to at the file writer level.